### PR TITLE
Fix error handling of AzureOpenAiChatModelListenerIT

### PR DESCRIPTION
**Pull request by qiaoleiatms**
## Issue
Fix comment in [PR#1305](https://github.com/langchain4j/langchain4j/pull/1305)

## Change
Ensure `StreamingResponseHandler.onError` been called while the response `FinishReason` is other than `CONTENT_FILTER` while any HttpResponseException is captured.
__This is a breaking change as previously all the HttpResponseException is treated as completed response instead of errored__

## General checklist

- [ ] There are no breaking changes
- [X] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new model integration

- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

**Comment by qiaoleiatms** [comment]
@langchain4j To continue the discussion of [PR#1305](https://github.com/langchain4j/langchain4j/pull/1305)
`ClientAuthenticationException` is a subclass of `HttpResponseException`, and previously it's been treated as a completed response.
Based on your input, it should be an errored response, so I made a change while the `FinishReason` is other than `CONTENT_FILTER`, the `onError` will be called.

@jdubois This change will be a breaking change on the behavior of error handling to the client users, need your input on this.

**Comment by jdubois** [comment]
Yes, that totally makes sense to me! Those errors should have sent an exception.
Concerning the tests, are you sure `String userMessage = "this message will fail";` is enough to make the test fail? I haven't tested it, I'm just surprised.

File: langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModelListenerIT.java [hunk 1]
```diff
@@ -46,8 +59,130 @@ protected Class<? extends Exception> expectedExceptionClass() {
         return ClientAuthenticationException.class;
     }
 
-    @Override
-    @Disabled("AzureOpenAiStreamingChatModel implementation is incorrect")
-    protected void should_listen_error() {
+    @Test
+    public void should_listen_request_and_response() {
```
Comment (langchain4j) [inline langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModelListenerIT.java]: No need to duplicate tests from `StreamingChatModelListenerIT`, you could just remove the method

**Review by ghost — COMMENTED** [review]
@qiaoleiatms thank you!

**Comment by langchain4j** [comment]
@jdubois `"this message will fail"` does not force the model to fail, but wrong API key does :)

File: langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModelListenerIT.java (hunk 1 above)
Comment (langchain4j) [reply langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModelListenerIT.java]: Once this is done, we can merge!

Comment (qiaoleiatms) [reply langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModelListenerIT.java]: Sure, already removed. 
Just noticed you move test cases to the super class

**Review by ghost — APPROVED** [review]
@qiaoleiatms thank you!

**Pull request merged**